src/main/java/de/srendi/advancedperipherals/common/blocks/base/PeripheralBlockEntity.java

```diff
@@ -82,6 +86,12 @@ public <T1> LazyOptional<T1> getCapability(@NotNull Capability<T1> cap, @Nullabl
                 handler = LazyOptional.of(() -> new SidedInvWrapper(this, Direction.NORTH));
             return handler.cast();
         }
+
+        if (cap == CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY && !remove && direction != null) {
+            if (fluidHandler == null || !fluidHandler.isPresent())
+                fluidHandler = FluidUtil.getFluidHandler(this.getLevel(), this.getBlockPos(), Direction.SOUTH);
```

Comment:
This leads to a StackOverflowError since `getFluidHandler` calls `getCapability` of the tile entity

Reply:
You might want to create a new fluid handler or want to use `net.minecraftforge.fluids.capability.TileFluidHandler`

Reply:
I created a new FluidTank. If I can ask, how I can implements with TileFluidHandler? I should replace 
`private LazyOptional<? extends IFluidHandler> fluidHandler;` with `private LazyOptional<? extends TileFluidHandler> fluidHandler;` ?

Reply:
The way with the fluid tank is simpler and more than enough, I just forgot that the fluid tank implements a fluid handler

Reply:
Ok thanks.